base/src/ast.rs

```diff
@@ -194,48 +194,48 @@ pub enum Pattern<Id: AstId> {
 
 #[derive(Clone, PartialEq, Debug)]
 pub struct Alternative<Id: AstId> {
-    pub pattern: LPattern<Id>,
-    pub expression: LExpr<Id>,
+    pub pattern: SPattern<Id>,
+    pub expression: SExpr<Id>,
 }
 
 #[derive(Clone, PartialEq, Debug)]
 pub struct Array<Id: AstId> {
     // Field to store the type of the array since type_of returns a borrowed reference
     pub id: Id,
-    pub expressions: Vec<LExpr<Id>>,
+    pub expressions: Vec<SExpr<Id>>,
 }
 
 #[derive(Clone, PartialEq, Debug)]
 pub struct Lambda<Id: AstId> {
     // Field to store the type of the array since type_of returns a borrowed reference
     pub id: Id,
     pub arguments: Vec<Id>,
-    pub body: Box<LExpr<Id>>,
+    pub body: Box<SExpr<Id>>,
 }
 
 /// Expression which contains a location
-pub type LExpr<Id> = Located<Expr<Id>>;
+pub type SExpr<Id> = Spanned<Expr<Id>>;
 
 #[derive(Clone, PartialEq, Debug)]
 pub enum Expr<Id: AstId> {
     Identifier(Id),
     Literal(LiteralEnum),
-    Call(Box<LExpr<Id>>, Vec<LExpr<Id>>),
-    IfElse(Box<LExpr<Id>>, Box<LExpr<Id>>, Option<Box<LExpr<Id>>>),
-    Match(Box<LExpr<Id>>, Vec<Alternative<Id>>),
-    BinOp(Box<LExpr<Id>>, Id, Box<LExpr<Id>>),
-    Let(Vec<Binding<Id>>, Box<LExpr<Id>>),
-    FieldAccess(Box<LExpr<Id>>, Id),
+    Call(Box<SExpr<Id>>, Vec<SExpr<Id>>),
+    IfElse(Box<SExpr<Id>>, Box<SExpr<Id>>, Option<Box<SExpr<Id>>>),
+    Match(Box<SExpr<Id>>, Vec<Alternative<Id>>),
+    BinOp(Box<SExpr<Id>>, Id, Box<SExpr<Id>>),
+    Let(Vec<Binding<Id>>, Box<SExpr<Id>>),
+    FieldAccess(Box<SExpr<Id>>, Id),
     Array(Array<Id>),
     Record {
         typ: Id,
         types: Vec<(Id::Untyped, Option<AstType<Id::Untyped>>)>,
-        exprs: Vec<(Id::Untyped, Option<LExpr<Id>>)>,
+        exprs: Vec<(Id::Untyped, Option<SExpr<Id>>)>,
     },
     Lambda(Lambda<Id>),
-    Tuple(Vec<LExpr<Id>>),
-    Type(Vec<TypeBinding<Id::Untyped>>, Box<LExpr<Id>>),
-    Block(Vec<LExpr<Id>>),
+    Tuple(Vec<SExpr<Id>>),
+    Type(Vec<TypeBinding<Id::Untyped>>, Box<SExpr<Id>>),
+    Block(Vec<SExpr<Id>>),
```

Comment:
I think this is a bit to many constructors.

```rust
pub enum Expr<Id> {
    Atom(Id),
    Cons(SExpr<Id>, SExpr<Id>),
}
```
Should be enough no? :stuck_out_tongue: